src/classes/UTIL_Profile_TEST.cls

```diff
@@ -0,0 +1,110 @@
+    verify: The query (Get Profiles) should run only once.
+    ***************************************************************************************************************************/   
+    private static testMethod void testGetInstanceSingleton() {
+    	Test.startTest();
+    	//Verify there is no query at the begining of the test.
+    	System.assertEquals(0, Limits.getQueries(), 'The query count should be 0 because no query has been run');
+
+    	UTIL_Profile profileUtil = UTIL_Profile.getInstance();
+    	//The GetInstance calls the UTIL_Profile Constructor which contains a query on the Profile.
+    	System.assertNotEquals(null, profileUtil, 'The GetInstance should return an instance of UTIL_Profile');
+    	System.assertEquals(1, Limits.getQueries(), 'The query count should be 1 because the GetInstance constructs the UTIL_Profile');
+
+    	//Test multiple calls of the GetInstance in a Bulk context.
+    	for (Integer i = 0; i < BULK_CALL; i++) {
+    		profileUtil = UTIL_Profile.getInstance();
+    	}
+    	//The GetInstance should not call the Constructor of UTIL_Profile anymore.
+    	System.assertNotEquals(null, profileUtil, 'The GetInstance should have the same instance of UTIL_Profile');
+    	System.assertEquals(1, Limits.getQueries(), 'The query count should not change');
+    	Test.stopTest();
+    }
+
+    /**************************************************************************************************************************
+    @description Test the GetProfileIds.
+    verify: The result returned by the GetProfileIds should is correct.
+    ***************************************************************************************************************************/   
+    private static testMethod void testGetProfileIds() {
+    	//Get all the Profiles in the Org, and group by Ids
+    	Map<String,List<Profile>> orgProfilesByName = getProfilesByName();
+
+    	Test.startTest();
+    	UTIL_Profile profileUtil = UTIL_Profile.getInstance();
+    	Test.stopTest();
+
+    	System.assertNotEquals(null, profileUtil, 'The GetInstance should return an instance of UTIL_Profile');
+
+    	for (String currentProfileName : orgProfilesByName.keySet()) {
+    		//Confirm that the result returned by GetProfileIds is correct.
+    		System.assertEquals(
+    			orgProfilesByName.get(currentProfileName).size(), 
+    			profileUtil.getProfileIds(currentProfileName).size(),
+    			'The number of Profiles that have the Name ' + currentProfileName + ' should be equals to the profileUtil.getProfileIds ones'
+    		);
+    		for (Profile currentProfile : orgProfilesByName.get(currentProfileName)) {
+    			System.assert(profileUtil.getProfileIds(currentProfileName).contains(currentProfile.Id));
+    		}
+    	}
+    }
+
+    /*********************************************************************************************************
+    * @description Get a Map of All Profiles, grouped by Name
+    * @return Map of Profiles, grouped by Name
```

Comment:
The `@return` should contain the data type and then optional description. For example: 

` * @return Map<String, List<Profile>> Profiles grouped by Name`

Also, note the space between data types in the Map.